pkg/vm/interop.go

```diff
@@ -1,13 +1,50 @@
 package vm
 
 import (
+	"crypto/sha256"
+	"encoding/binary"
 	"errors"
 	"fmt"
+	"sort"
 )
 
 // InteropFunc allows to hook into the VM.
 type InteropFunc func(vm *VM) error
 
+// InteropFuncPrice represents an interop function with a price.
+type InteropFuncPrice struct {
+	Func  InteropFunc
+	Price int
+}
+
+// interopIDFuncPrice adds an ID to the InteropFuncPrice.
+type interopIDFuncPrice struct {
+	ID uint32
+	InteropFuncPrice
+}
+
+// InteropGetterFunc is a function that returns an interop function-price
+// structure by the given interop ID.
+type InteropGetterFunc func(uint32) *InteropFuncPrice
+
+var defaultVMInterops []interopIDFuncPrice
+
+func getDefaultVMInterop(id uint32) *InteropFuncPrice {
+	n := sort.Search(len(defaultVMInterops), func(i int) bool {
+		return defaultVMInterops[i].ID >= id
+	})
+	if n < len(defaultVMInterops) && defaultVMInterops[n].ID == id {
+		return &defaultVMInterops[n].InteropFuncPrice
+	}
+	return nil
+}
+
+// InteropNameToID returns an identificator of the method based on its name.
+func InteropNameToID(name []byte) uint32 {
```

Comment:
hm, walked through usages and it looks like it's better to accept `name string ` instead of `name []byte`, wdyt?

Reply:
The most often usecase is in the `SYSCALL` processing and that's the place where I want to avoid type conversions (and associated slice creations). All the other ones are either in `init()` or in tests, it's much cheaper to do conversions there.